pkg/skaffold/build/buildpacks/buildpacks.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package buildpacks
+
+import (
+	"context"
+	"io"
+
+	"github.com/pkg/errors"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+)
+
+// BuildpackBuilder is a builder for buildpack artifacts
+type BuildpackBuilder struct {
+	localDocker docker.LocalDaemon
+	pushImages  bool
+}
+
+// NewArtifactBuilder returns a new buildpack artifact builder
+func NewArtifactBuilder(localDocker docker.LocalDaemon, pushImages bool) *BuildpackBuilder {
+	return &BuildpackBuilder{
+		localDocker: localDocker,
+		pushImages:  pushImages,
+	}
+}
+
+// Build builds an artifact with the pack CLI.
+func (b *BuildpackBuilder) Build(ctx context.Context, out io.Writer, a *latest.Artifact, tag string) (string, error) {
+	// To improve caching, we always build the image with [:latest] tag
+	// This way, the lifecycle is able to "bootstrap" from the previously built image.
+	parsed, err := docker.ParseReference(tag)
+	if err != nil {
+		return "", errors.Wrapf(err, "parsing tag %s", tag)
+	}
+	latest := parsed.BaseName + ":latest"
+
+	if err := b.build(ctx, out, a.Workspace, a.BuildpackArtifact, latest); err != nil {
+		return "", err
+	}
+
+	if err := b.localDocker.Tag(ctx, latest, tag); err != nil {
```

Comment:
ahh, good to see `latest` being used in the proper way :) we should definitely document this since it's a bit different than the normal skaffold tagging mechanism.